source/grid/grid_tools.cc

```diff
@@ -3257,24 +3289,31 @@ namespace GridTools
       for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
         if (cell->face(f)->at_boundary())
           {
-            // copy boundary to manifold ids
-            cell->face(f)->set_manifold_id
-            (static_cast<types::manifold_id>(cell->face(f)->boundary_id()));
+            auto bid = cell->face(f)->boundary_id();
+            auto ind = std::find(src_boundary_ids.begin(), src_boundary_ids.end(), bid)-
+                       src_boundary_ids.begin();
 
-            // then reset boundary ids if so desired, and in 3d also that
-            // of edges
-            if (reset_boundary_ids == true)
+            if ((unsigned int)ind < src_boundary_ids.size())
               {
-                cell->face(f)->set_boundary_id(0);
-                if (dim >= 3)
-                  for (signed int e=0; e<static_cast<signed int>(GeometryInfo<dim>::lines_per_face); ++e)
-                    cell->face(f)->line(e)->set_boundary_id(0);
+                // assign the manifold id
+                cell->face(f)->set_manifold_id(dst_manifold_ids[ind]);
+                // then reset boundary id
+                cell->face(f)->set_boundary_id(reset_boundary_ids[ind]);
               }
+
+            if (dim >= 3)
+              for (signed int e=0; e<static_cast<signed int>(GeometryInfo<dim>::lines_per_face); ++e)
+                {
+                  auto bid = cell->face(f)->line(e)->boundary_id();
+                  auto ind = std::find(src_boundary_ids.begin(), src_boundary_ids.end(), bid)-
+                             src_boundary_ids.begin();
+                  if ((unsigned int)ind < src_boundary_ids.size())
+                    cell->face(f)->line(e)->set_boundary_id(reset_boundary_ids[ind]);
```

Comment:
Am I correct that for an edge shared by faces with different boundary id's then its not really possible to predetermine what the resulting ID on the edge will be? If so then maybe its best if we document this.

Reply:
Mmmmh. Why would it be so? The edge has its own boundary id, independent on the face. Here I'm resetting it if it is in the list of ids to reset...

Reply:
Yes, of course. My mistake!